### PR TITLE
refactor(api-markdown-documenter): Add `{@link}` tag reference validation to linter

### PR DESCRIPTION
**Pull request by Josmithr**
Follow-up to #22150. Adds validation for `{@link}` references within the model.

### CODE REVIEW

File: tools/api-markdown-documenter/src/test/LintApiModel.test.ts [hunk 1]
```diff
@@ -6,24 +6,52 @@
...
+
+		expect(result).to.be.undefined;
+	});
+
+	it("Empty API Model yields no errors", async () => {
+		const apiModel = new ApiModel();
+		const result = await lintApiModel({ apiModel });
+
+		expect(result).to.be.undefined;
+	});
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/test/LintApiModel.test.ts:31]:
```suggestion
```

File: tools/api-markdown-documenter/src/LintApiModel.ts [hunk 2]
```diff
@@ -76,12 +85,12 @@ interface MutableLinterErrors {
  * Errors found during linting.
  */
 export interface LinterErrors {
+	// TODO: malformed tag errors
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/LintApiModel.ts:88]: Future PR

File: tools/api-markdown-documenter/src/LintApiModel.ts [hunk 3]
```diff
@@ -156,6 +168,135 @@ function lintApiItem(
...
+		case DocNodeKind.LinkTag: {
+			const result = checkLinkTag(node as DocLinkTag, associatedItem, apiModel);
+			if (result !== undefined) {
+				errors.referenceErrors.add(result);
+			}
+			break;
+		}
+		case DocNodeKind.InheritDocTag: {
+			fail(
+				"Encountered an @inheritDoc tag while walking a TSDoc tree. API-Extractor resolves such tags at a higher level, so this is unexpected.",
```
Comment (TommyBrosman) [inline tools/api-markdown-documenter/src/LintApiModel.ts:244]: I don't understand this failure message. Can you link to the documentation? I'm not sure what it resolving at a "higher level" means.

File: tools/api-markdown-documenter/src/LintApiModel.ts [hunk 4]
```diff
@@ -156,6 +168,135 @@ function lintApiItem(
...
+	assert(
+		linkTag.codeDestination !== undefined,
+		"Expected a `codeDestination` or `urlDestination` to be defined, but neither was.",
+	);
+
+	// If the link is a symbolic reference, validate it.
+	try {
+		resolveSymbolicReference(apiItem, linkTag.codeDestination, apiModel);
+	} catch (error: unknown) {
+		assert(error instanceof Error, "Expected an error.");
```
Comment (TommyBrosman) [inline tools/api-markdown-documenter/src/LintApiModel.ts:287]: Is this assert needed?

File: tools/api-markdown-documenter/src/LintApiModel.ts [hunk 5]
```diff
@@ -156,6 +168,135 @@ function lintApiItem(
...
+	for (const node of nodes) {
+		checkTagsUnderTsdocNode(node, associatedItem, apiModel, errors);
+	}
+}
+
+function checkLinkTag(
+	linkTag: DocLinkTag,
+	apiItem: ApiItem,
+	apiModel: ApiModel,
+): ReferenceError | undefined {
```
Comment (TommyBrosman) [inline tools/api-markdown-documenter/src/LintApiModel.ts:268]: Can you document what the return type indicates when it is undefined?

File: tools/api-markdown-documenter/src/LintApiModel.ts (hunk 4 above)
Comment (Josmithr) [reply tools/api-markdown-documenter/src/LintApiModel.ts:287]: Ha, I guess not. 
```suggestion
```

File: tools/api-markdown-documenter/src/LintApiModel.ts (hunk 3 above)
Comment (Josmithr) [reply tools/api-markdown-documenter/src/LintApiModel.ts:258]: I added a comment pointing to existing notes I left in `lintApiItem`. Hopefully that sufficiently clarifies things.

**Review by TommyBrosman — APPROVED** [review]

**Pull request merged**